opnode/test/system_test.go

```diff
@@ -26,206 +24,174 @@ import (
+				L2EngineAddrs: []string{"ws://127.0.0.1:9091"},
+				L2NodeAddr:    "ws://127.0.0.1:9091",
+				L1TrustRPC:    false,
 			},
-			ethConfig: &ethconfig.Config{
-				NetworkId: 901,
+			"sequencer": {
+				L1NodeAddr:    "ws://127.0.0.1:9090",
+				L2EngineAddrs: []string{"ws://127.0.0.1:9092"},
+				L2NodeAddr:    "ws://127.0.0.1:9092",
+				L1TrustRPC:    false,
+				Sequencer:     true,
+				// Submitter PrivKey is set in system start for rollup nodes where sequencer = true
+				RPCListenAddr: "127.0.0.1",
+				RPCListenPort: 9093,
 			},
 		},
-		l2Sequencer: gethConfig{
-			nodeConfig: &node.Config{
-				Name:   "l2gethSeq",
-				WSHost: "127.0.0.1",
-				WSPort: 9092,
-			},
-			ethConfig: &ethconfig.Config{
-				NetworkId: 901,
-			},
+		Loggers: map[string]log.Logger{
+			"verifier":  testlog.Logger(t, log.LvlError),
+			"sequencer": testlog.Logger(t, log.LvlError),
 		},
-		premine: map[string]int{
-			"m/44'/60'/0'/0/0": 10000000,
-			"m/44'/60'/0'/0/1": 10000000,
-			"m/44'/60'/0'/0/2": 10000000,
-			l2OutputHDPath:     10000000,
-			bssHDPath:          10000000,
+		RollupConfig: rollup.Config{
+			BlockTime:         1,
+			MaxSequencerDrift: 10,
+			SeqWindowSize:     2,
+			L1ChainID:         big.NewInt(900),
+			// TODO pick defaults
+			FeeRecipientAddress: common.Address{0xff, 0x01},
+			BatchInboxAddress:   common.Address{0xff, 0x02},
+			// Batch Sender address is filled out in system start
 		},
-		cliqueSigners:          []string{"m/44'/60'/0'/0/0"},
-		depositContractAddress: derive.DepositContractAddr.Hex(),
-		l1InfoPredeployAddress: derive.L1InfoPredeployAddr.Hex(),
 	}
-	// Create genesis & assign it to ethconfigs
-	initializeGenesis(cfg)
-
-	// Start L1
-	l1Node, l1Backend, err := l1Geth(cfg)
-	require.Nil(t, err)
-	defer l1Node.Close()
-
-	err = l1Node.Start()
-	require.Nil(t, err)
+}
 
-	err = l1Backend.StartMining(1)
-	require.Nil(t, err)
+func waitForTransaction(hash common.Hash, client *ethclient.Client, pollInterval, timeout time.Duration) (*types.Receipt, error) {
+	timeoutCh := time.After(timeout)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	for {
+		receipt, err := client.TransactionReceipt(ctx, hash)
+		if receipt != nil && err == nil {
+			return receipt, nil
+		} else if err != nil && !errors.Is(err, ethereum.NotFound) {
+			return nil, err
+		}
 
-	l1Client, err := ethclient.Dial(endpoint(cfg.l1.nodeConfig))
-	require.Nil(t, err)
-	l1GenesisID, _ := getGenesisInfo(l1Client)
+		select {
+		case <-timeoutCh:
+			return nil, errors.New("timeout")
+		case <-time.After(pollInterval):
+		}
+	}
+}
 
-	// Start L2
-	l2Node, _, err := l2Geth(cfg)
-	require.Nil(t, err)
-	defer l2Node.Close()
+// waitForDepositOnL2 waits for a deposit by polling an account for the change in balance.
+// It returns the block on which the change occurred.
+func waitForDepositOnL2(addr common.Address, mint *big.Int, client *ethclient.Client, timeout time.Duration) (uint64, error) {
```

Comment:
We can now wait for a deposit the same way as we wait for any tx: we first derive the L2 tx from the L1 receipt, compute the tx-hash, and then poll for confirmation of that. See https://github.com/ethereum-optimism/optimistic-specs/pull/351/commits/1467d368d85b1c95eecda8ceec43214454d7e673 for an example